tests/unit/lib/telemetry/test_container_engine_metrics.py

```diff
@@ -26,86 +27,48 @@ class TestContainerEngineMetrics(TestCase):
     def setUp(self):
         self.mock_ctx = MockContext()
 
-    @patch("samcli.cli.context.Context.get_current_context")
+    @parameterized.expand(
+        [
+            # (socket_path, expected_engine, test_description)
+            ("unix://~/.finch/finch.sock", "finch", "Finch container engine from global storage"),
+            ("unix://~/.rd/docker.sock", "rancher-desktop", "Rancher Desktop container engine from global storage"),
+        ]
+    )
+    @patch("samcli.lib.telemetry.metric.get_container_runtime_telemetry_info")
     @patch("samcli.lib.telemetry.metric.Telemetry")
     @patch("samcli.lib.telemetry.event.EventTracker.send_events")
     @patch("samcli.lib.telemetry.metric.get_all_experimental_statues")
     @patch("samcli.lib.telemetry.metric.get_git_remote_origin_url")
     @patch("samcli.lib.telemetry.metric.get_project_name")
     @patch("samcli.lib.telemetry.metric.get_initial_commit_hash")
-    def test_container_engine_from_context_finch(
+    def test_container_engine_from_global_storage_scenarios(
         self,
+        socket_path,
+        expected_engine,
+        description,
```

Comment:
We don't really need the `description` here just to show a nicer error message if it fails. The test name and values are enough.